### PR TITLE
guide: Spring Boot to Micronaut Framework - Application class

### PR DESCRIPTION
**Pull request by sdelamo**
This PR adds the ability to generate Spring Boot applications. It is currently limited to Gradle, but we can expand it to Maven as well.

This PR also adds a simple guide which has two sample applications

- `micronautframework`. A Micronaut Application

-`springboot`. A Spring Boot application

### CODE REVIEW

**Comment by graemerocher** [comment]
does it work with Maven?

**Comment by sdelamo** [comment]
> does it work with Maven?

Yes

**Review by timyates — COMMENTED** [review]
The spring boot apps still contain a src/main/resources/application.yml file containing

```
micronaut:
  application:
    name: micronautguide
```

Should this be removed?

File: buildSrc/src/main/java/io/micronaut/guides/feature/springboot/replacements/MavenReplacement.java [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+    }
+
+    protected void addPom(GeneratorContext generatorContext) {
+        MavenBuild mavenBuild = createBuild(generatorContext);
+        generatorContext.addTemplate("mavenPom", new RockerTemplate("pom.xml", pom(generatorContext, mavenBuild)));
+    }
+
+    protected RockerModel pom(GeneratorContext generatorContext, MavenBuild mavenBuild) {
+        if (SpringBootApplicationFeature.isSpringBootApplication(generatorContext)) {
+            Coordinate springBootParent = coordinateResolver.resolve("spring-boot-starter-parent").orElseThrow();
```
Comment (timyates) [inline buildSrc/src/main/java/io/micronaut/guides/feature/springboot/replacements/MavenReplacement.java:93]: This `orElseThrow` is a Java 11+ method, but I guess that's fine

**Review by timyates — APPROVED** [review]
An alternative would be to call start.spring.io, unpack, and copy over our custom classes?

**Comment by sdelamo** [comment]
> The spring boot apps still contain a src/main/resources/application.yml file containing
> 
> ```
> micronaut:
>   application:
>     name: micronautguide
> ```
> 
> Should this be removed?

There are still slight details such as the one you are describing. 

For example, Spring boot apps are generated in the `example.micronaut` package

However, we can improve those details in a separate PR.

**Pull request merged**